src/main/java/org/openrewrite/java/testing/junit5/TemporaryFolderToTempDir.java

```diff
@@ -93,187 +100,181 @@ protected TemporaryFolderToTempDirVisitor getVisitor() {
                     m.getCoordinates().replace(),
                     modifiers,
                     fieldVars);
-
-            doAfterVisit(new ReplaceTemporaryFolderMethods(m));
             return m;
         }
 
-        /**
-         * This visitor replaces methods from TemporaryFolder with JUnit5-compatible alternatives:
-         * <p>
-         * File getRoot()
-         * File newFile()
-         * File newFile(String fileName)
-         * File newFolder()
-         * File newFolder(String... folderNames)
-         * File newFolder(String folder)
-         */
-        private static class ReplaceTemporaryFolderMethods extends JavaVisitor<ExecutionContext> {
-
-            private final J.VariableDeclarations tempDirFields;
-
-            private ReplaceTemporaryFolderMethods(J.VariableDeclarations tempDirFields) {
-                this.tempDirFields = tempDirFields;
+        @Override
+        public J visitNewClass(J.NewClass newClass, ExecutionContext executionContext) {
+            J.NewClass nc = (J.NewClass) super.visitNewClass(newClass, executionContext);
+            JavaType.FullyQualified fq = TypeUtils.asFullyQualified(nc.getType());
+            if (fq != null && "org.junit.rules.TemporaryFolder".equals(fq.getFullyQualifiedName())) {
+                return null;
```

Comment:
not sure I understand this, are you trying to clear the initializer for a named variable here? 

This would also impact places where `new org.junit.rules.TemporaryFolder()` is used in code (which probably isn't very often)

Reply:
Yes, the goal is to clear `TemporaryFolder` initializers, then annotate their associated variables with `@TempDir` and finally change `TemporaryFolder` types to `File`.